src/scales/scale.time.js

```diff
@@ -7,10 +7,132 @@ moment = typeof(moment) === 'function' ? moment : window.moment;
 var defaults = require('../core/core.defaults');
 var helpers = require('../helpers/index');
 
+function sorter(a, b) {
+	return a - b;
+}
+
+/**
+ * Returns an array of {time, pos} objects used to interpolate a specific `time` or position
+ * (`pos`) on the scale, by searching entries before and after the requested value. `pos` is
+ * a decimal between 0 and 1: 0 being the start of the scale (left or top) and 1 the other
+ * extremity (left + width or top + height). Note that it would be more optimized to directly
+ * store pre-computed pixels, but the scale dimensions are not guaranteed at the time we need
+ * to create the lookup table. The table ALWAYS contains at least two items: min and max.
+ *
+ * @param {Number[]} timestamps - timestamps sorted from lowest to highest.
+ * @param {Boolean} linear - If true, timestamps will be spread linearly along the min/max
+ * range, so basically, the table will contains only two items: {min, 0} and {max, 1}. If
+ * false, timestamps will be positioned at the same distance from each other. In this case,
+ * only timestamps that break the time linearity are registered, meaning that in the best
+ * case, all timestamps are linear, the table contains only min and max.
+ */
+function buildLookupTable(timestamps, min, max, linear) {
+	if (linear || !timestamps.length) {
+		return [
+			{time: min, pos: 0},
+			{time: max, pos: 1}
+		];
+	}
+
+	var table = [];
+	var items = timestamps.slice(0);
+	var i, ilen, prev, curr, next;
+
+	if (min < timestamps[0]) {
+		items.unshift(min);
+	}
+	if (max > timestamps[timestamps.length - 1]) {
+		items.push(max);
+	}
+
+	for (i = 0, ilen = items.length; i<ilen; ++i) {
+		next = items[i + 1] || 0;
+		prev = items[i - 1] || 0;
```

Comment:
this `|| 0` gives me pause

Let's say you have:
* `i = 0`
* `prev = 0`
* `curr = 2`
* `next = 4`

Or you have:
* `i = 0`
* `prev = 0`
* `curr = 2`
* `next = 5`

Is it really the case that one of these should trigger the `if` statement and one shouldn't? I guess in actuality that's quite unlikely to happen, but it's probably slightly safer to leave the variable `undefined` and then add an extra check for that in the `if` statement